### PR TITLE
M3-5529: Update "Add an IP Address" drawer and provide for ability to add IPv6 ranges in it

### PR DESCRIPTION
**Pull request by DevDW**
## Description
Updates the types and requests to support the new IPv6 functionality, updates the mocks accordingly, and makes a few changes in the "Add an IP Address" drawer:

- Changes IPv4 dropdown to radio buttons
- Adds IPv6 radio buttons

## To-Do
- [x] Clean up & refactor `<AddIPDrawer />`
- [x] Final description copy

## How to test
From the "Network" tab of a Linode Detail page, click "Add an IP Address" and confirm that no functionality has been lost for allocating IPv4 addresses and that you can now allocate IPv6 addresses successfully.

### CODE REVIEW

File: packages/manager/src/features/linodes/LinodesDetail/LinodeNetworking/AddIPDrawer.tsx [hunk 1]
```diff
@@ -122,29 +193,43 @@ const AddIPDrawer: React.FC<CombinedProps> = (props) => {
   );
 
   const _tooltipCopy =
-    disabled && selected
+    disabledIPv4 && selectedIPv4
       ? readOnly
         ? 'You do not have permission to modify this Linode.'
-        : tooltipCopy[selected]
+        : tooltipCopy[selectedIPv4]
       : null;
```
Comment (codyfinn) [inline packages/manager/src/features/linodes/LinodesDetail/LinodeNetworking/AddIPDrawer.tsx:200]: Is it possible to un-nest this logic?

Comment (DevDW) [reply packages/manager/src/features/linodes/LinodesDetail/LinodeNetworking/AddIPDrawer.tsx:177]: Do you think breaking it up into some `if` statements would be easier to follow?

**Review by johnwcallahan — COMMENTED** [review]
I know this is still WIP, but I had a quick look and the general approach looks good to me.

File: packages/manager/src/features/linodes/LinodesDetail/LinodeNetworking/AddIPDrawer.tsx (hunk 1 above)
Comment (codyfinn) [reply packages/manager/src/features/linodes/LinodesDetail/LinodeNetworking/AddIPDrawer.tsx:177]: That would probably help make it more readable. I'm more focused on the nested logic in general though.

**Review by HanaXu — COMMENTED** [review]
Thoughts on adding a success toast after allocation? 🤔

**Comment by DevDW** [comment]
> Thoughts on adding a success toast after allocation? 🤔

I think that would be fine but I'm not sure how necessary it is since the allocation happens quickly and the user remains on the same page where the new IP shows up

File: packages/manager/src/features/linodes/LinodesDetail/LinodeNetworking/AddIPDrawer.tsx (hunk 1 above)
Comment (DevDW) [reply packages/manager/src/features/linodes/LinodesDetail/LinodeNetworking/AddIPDrawer.tsx:197]: Going to opt to keep it as is for now unless anyone feels strongly about it; I don't think the nesting is too bad here

**Review by HanaXu — APPROVED** [review]
🚀

**Pull request merged**